proxy/socks5.go

```diff
@@ -26,14 +26,21 @@ type Socks5 struct {
 }
 
 func NewSocks5(addr, user, pass string) (*Socks5, error) {
+	unix := len(addr) > 0 && addr[0] == '/'
```

Comment:
```suggestion
	unix := len(addr) > 0 && (addr[0] == '/' || addr[0] == '@' || addr[0] == 0x00)
```
Just wondering why not ^?

Reply:
You pass the url.path from url instance when url.host is empty. If we use the format like socks5:///@test, the address passed to NewSocks will be /@test, which requires additional processing logic. and for len(address)>2, it is only for a more safe in case the out of index error.  The approach here checks if the first char after / is @ or null(0x00), and remove the leading /, since it will cause error.

Reply:
And the reason we can’t avoid using socks5:/// is the string before @ will be parsed as username/password whenever is has valid user:pass or not. The @ char will be omitted by parser.

Reply:
Oh I see. that makes sense. But I think such hack adjustment might be better to add here in the `parseSocks5` function:

https://github.com/xjasonlyu/tun2socks/blob/ec854101442986cb7390a0c82f25992978a6c7b1/engine/parse.go#L123-L132

And `NewSocks5` func should keep its straightforwardness?

Reply:
It makes sense, in this case, the unix check logic may also move to parser function as well. And I will need to make a new arg for unix flag for NewSocks5 , else it will need redundant code to test if the address is unix or not.

Reply:
Yeah, it sounds good